packages/ckeditor5-image/src/image/imageediting.js

```diff
@@ -58,23 +61,49 @@ export default class ImageEditing extends Plugin {
 			allowWhere: '$block',
 			allowAttributes: [ 'alt', 'src', 'srcset' ]
 		} );
-
-		conversion.for( 'dataDowncast' ).elementToElement( {
-			model: 'image',
-			view: ( modelElement, { writer } ) => createImageViewElement( writer )
+		schema.register( 'imageInline', {
+			isObject: true,
+			isInline: true,
+			allowWhere: '$text',
+			allowAttributes: [ 'alt', 'src', 'srcset' ]
 		} );
 
-		conversion.for( 'editingDowncast' ).elementToElement( {
-			model: 'image',
-			view: ( modelElement, { writer } ) => toImageWidget( createImageViewElement( writer ), writer, t( 'image widget' ) )
-		} );
+		conversion.for( 'dataDowncast' )
+			.elementToElement( {
+				model: 'image',
+				view: ( modelElement, { writer } ) => createImageViewElement( writer )
+			} )
+			.elementToElement( {
+				model: 'imageInline',
+				view: ( modelElement, { writer } ) => writer.createEmptyElement( 'img' )
+			} );
+
+		conversion.for( 'editingDowncast' )
+			.elementToElement( {
+				model: 'image',
+				view: ( modelElement, { writer } ) => toImageWidget( createImageViewElement( writer ), writer, t( 'image widget' ) )
+			} )
+			.elementToElement( {
+				model: 'imageInline',
+				view: ( modelElement, { writer } ) => {
+					const emptyElement = writer.createEmptyElement( 'img' );
+					const element = writer.createContainerElement( 'span' );
+					writer.insert( writer.createPositionAt( element, 0 ), emptyElement );
```

Comment:
```suggestion
					const imgElement = writer.createEmptyElement( 'img' );
					const wrapperElement = writer.createContainerElement( 'span' );
					writer.insert( writer.createPositionAt( wrapperElement, 0 ), imgElement );
```

Also it would be good to add some class name to that `span`.